build/builder.go

```diff
@@ -244,7 +259,8 @@ type Builder struct {
 	todo         []*zoekt.Document
 	size         int
 
-	parser ctags.Parser
+	parser     ctags.Parser
+	scipParser ctags.Parser
```

Comment:
can we introduce a parser which given a languagemap will pick the correct parser to use? That way we can hide most of this logic away from indexers.

Reply:
Looking into this some more, we can't use the goctags Parser interface or else we'd have to redetermine the language on parse every time. Not sure if it's worth it modifying the parser like this directly.

Reply:
yeah. I reckon you don't need to make a wrapper. But maybe given the extra complexity here you create a new thing which given enablectags, enablescipctags, languagemap returns something which mutates a zoekt.Document (something which likely looks a lot like func ctagsAddSymbolsParser?).

I see we still have code paths that don't go via parser. I don't think those code paths are exercised in practice?

Reply:
Unsure, they're for using exuberant-ctags which I believe is a supported ctags for sourcegraph/sourcegraph(?), not sure anyone uses it though